returnDetail.go

```diff
@@ -216,13 +216,20 @@ func (rd *ReturnDetail) Parse(record string) {
 func (rd *ReturnDetail) UnmarshalJSON(data []byte) error {
 	type Alias ReturnDetail
 	aux := struct {
+		// json.RawMessage is used here to allow library to still parse json files that stored
+		// RNI in an int. json.Number could also be used, but X937 standard doesn't specify a
+		// type for the RNI, so, RawMessage provides flexibility to handle non-numeric values
```

Comment:
Hey @lvanoort, thanks for the PR!

The doc I have says ReturnNotificationIndicator is numeric and can be empty, `1` or `2`:
![image](https://user-images.githubusercontent.com/11085638/101414134-c1ab0480-38a2-11eb-9f15-fec4f1eac751.png)

Maybe `json.Number` would be better?

Reply:
@atonks2 looks like my doc is a bit incomplete then (it pretty much says nothing about ReturnNotificationIndicator's content is besides indicating that it's not validated by a lot of exchange networks). 

It turns out that Go changed the behaviour of `json.Number` in 1.14 to be stricter about its contents, making it no longer capable of doing what we're after here, so `json.RawMessage` is still probably the way to go. That said, I'll add a touch more validation to `UnmarshalJSON` so it only allows numbers, quoted numbers, and empty values (and get rid of the comment about unspecified type).

Reply:
That works for me. Thanks again for the fix!